app/poi_api/poi_service.py

```diff
@@ -0,0 +1,106 @@
+from app import firestore_db
+from flask import jsonify
+from .poi_api import POIApi
+from .poi_suggestion import POI_suggestions
+from .poi import POI
+from .poi_errors import POINotFoundError, InvalidPOISuggestionError
+from datetime import datetime
+
+
+class POI_Service(POIApi):
+    def get_all_POI(self):
+        try:
+            poi_ref = firestore_db.collection("POI")
+            all_poi = [doc.to_dict() for doc in poi_ref.stream()]
+            list_all_poi = []
+            for poi in all_poi:
+                poi_object = self._create_poi_object(poi)
+                list_all_poi.append(poi_object)
+            return list_all_poi
+        except Exception as e:
+            raise Exception(f"An error occured: {e}")
+
+    def get_POI(self, poi_id):
+        try:
+            poi_ref = firestore_db.collection("POI")
+            poi = poi_ref.document(poi_id).get()
+            poi_dict = poi.to_dict()
+            if poi_dict is not None:
+                return self._create_poi_object(poi_dict)
+            else:
+                raise POINotFoundError(
+                    f"The POI with the id: {poi_id} could not be found", 404
+                )
+        except POINotFoundError as e:
+            raise POINotFoundError(
+                f"The POI with the id: {poi_id} could not be found", 404
+            )
+        except Exception as e:
+            raise Exception(f"An error occured: {e}", 404)
+
+    def suggest_new_POI(self, poi_suggestion):
+        try:
+            poi_suggestion_ref = firestore_db.collection("POI_proposal").document()
+            # Generate id for poi suggestion document
+            pid = poi_suggestion_ref.id
+            poi_suggestion_instance = self._create_POI_suggestion(pid, poi_suggestion)
+            self._save_POI_suggestion(poi_suggestion_instance, poi_suggestion_ref)

```

Comment:
Is this a conversion from firestore doc to POI back to firestore doc just to save it. If it's not necessary you don't have to convert back and forth and just save it directly.